acinclude/krb5.m4

```diff
@@ -8,6 +8,51 @@
 dnl these checks must be performed in the same order as here defined,
 dnl and have mostly been lifted out of an inlined configure.ac.
 
+AC_DEFUN([SQUID_CHECK_SOLARIS_KRB5],[
+  # no pkg-config for solaris native Kerberos
+  AS_IF([test "$cross_compiling" = "no" -a "x$with_mit_krb5" != "xyes" -a "x$with_mit_krb5" != "xno"],[
+    AC_PATH_PROG(krb5_config,krb5-config,no,[$PATH:$with_mit_krb5/bin])
+  ])
+  AC_MSG_NOTICE([Use krb5-config to get CFLAGS and LIBS])
+  LIBMIT_KRB5_CFLAGS="-I/usr/include/kerberosv5 `$ac_krb5_config --cflags krb5 2>/dev/null`"
+  LIBMIT_KRB5_LIBS="`$ac_krb5_config --libs krb5 2>/dev/null`"
+  # Solaris 10 Update 11 patches the krb5-config tool to produce stderr messages on stdout.
+  SOLARIS_BROKEN_KRB5CONFIG_GSSAPI="`$ac_krb5_config --libs gssapi 2>/dev/null | grep "krb5-config"`"
+  AS_IF([test "x$SOLARIS_BROKEN_KRB5CONFIG_GSSAPI" = "x"],[
+    LIBMIT_KRB5_CFLAGS="$LIBMIT_KRB5_CFLAGS `$ac_krb5_config --cflags gssapi 2>/dev/null`"
+    LIBMIT_KRB5_LIBS="$LIBMIT_KRB5_LIBS `$ac_krb5_config --libs gssapi 2>/dev/null`"
+  ])
+  CPPFLAGS="$LIBMIT_KRB5_CFLAGS $CPPFLAGS"
+  LIBS="$LIBMIT_KRB5_LIBS $LIBS"
+  missing_required=no
+  AC_MSG_NOTICE([Try to find Kerberos libraries in given path])
+  AC_CHECK_LIB(krb5, [main], [LIBMIT_KRB5_LIBS="$LIBMIT_KRB5_LIBS -lkrb5"],[
+    AC_MSG_WARN([library 'krb5' is required for Solaris Kerberos])
+    missing_required=yes
+  ])
+  AC_CHECK_LIB(gss, [main], [LIBMIT_KRB5_LIBS="$LIBMIT_KRB5_LIBS -lgss"],[
```

Comment:
Can we reuse the GSS check that is done elsewhere? The fact that two checks are different is a red flag. If they have to be different, it would be good to add a source code comment to explain why.

Reply:
FFWIW I suspect we can probably do without this bit and adjust the `--with-gss` logic to allow detecting it on Solaris. 

However, I/we lack a Solaris machine to test out logic changes like that properly. So for now I am leaving this as-is.

Reply:
> we lack a Solaris machine to test out logic changes like that properly. 

... and yet this PR is changing Solaris-specific code. In this context, I see no substantial difference between the untested changes already done and the untested changes my request implies. IMO, in this case, it is best to stop propagating problematic Solaris-specific code that we cannot test then to modify it without testing.